libarchive/archive_read_support_format_zip.c

```diff
@@ -1174,17 +1227,45 @@ zip_read_local_file_header(struct archive_read *a, struct archive_entry *entry,
 			return ARCHIVE_FATAL;
 		}
 
+    size_t linkname_full_length = linkname_length; // take into account compression if any
+    if (zip->entry->compression != 0) // symlink target string appeared to be compressed
+    {
+      int status = ARCHIVE_FATAL;
+      char *uncompressed_buffer = (char*) malloc(zip_entry->uncompressed_size);
+      if (uncompressed_buffer == NULL)
+      {
+        archive_set_error(&a->archive, ENOMEM, "No memory for lzma decompression");
+        return status;
+      }
+
+      switch (zip->entry->compression)
+      {
+#if HAVE_LZMA_H && HAVE_LIBLZMA
+        case 14: /* ZIPx LZMA compression. */
+          status = zipx_lzma_uncompress_buffer(p, linkname_length, uncompressed_buffer, (size_t)zip_entry->uncompressed_size);
+          break;
+#endif
+        default: /* Unsupported compression. */
+          break;
+      }
+      if (status == ARCHIVE_OK)
+      {
+        p = uncompressed_buffer;
+        linkname_full_length = (size_t)zip_entry->uncompressed_size;
+      }
```

Comment:
If the LZMA support is not compiled in, or LZMA decompression failed, shouldn't this be an error condition? It looks like in case of missing LZMA support, zipx unpacker creates symbolic links which point to invalid filenames.

Maybe report an error here? Something like:

```c
          if (status == ARCHIVE_OK)
          {
              ....
          }
          else
          {
              archive_set_error(&a->archive, ARCHIVE_ERRNO_FILE_FORMAT,
                            "Unsupported ZIP compression method during decompression of link entry (%d: %s)",
                            zip->entry->compression, compression_name(zip->entry->compression));
              return ARCHIVE_FAILED;
          }
```

Reply:
agree. added